### DISCUSSION TITLE
X96 emmc

### DISCUSSION THREAD

**Discussion by ratsalad** [devmfc/debian-on-amlogic#39] [Q&A]
Seem to have the same issue with a X96 (950X) box. 

I tried with slimbox installed and then reverted back to stock, but neither worked.

tvbox:~#./aml-multiboot-setup.sh Configuration file wrong or corrupted
[Error] 
fw_printenv has no valid config file

Found valid Amlogic Partition Table with 14 partitions
======================================================================
0 MiB --- offset: 0x0 B --- 0 B --- 0 s ---
	name:	bootloader
	flags:	0
	size:	4 MiB 	0x400000 B  
	end:	4 MiB 	0x400000 B  
  ---------------------
  ####  GAP 32 MiB  ####
36 MiB --- offset: 0x2400000 B --- 37748736 B --- 73728 s ---
	name:	reserved
	flags:	0
	size:	64 MiB 	0x4000000 B  
	end:	100 MiB 	0x6400000 B  
  ---------------------
  ####  GAP 8 MiB  ####
108 MiB --- offset: 0x6c00000 B --- 113246208 B --- 221184 s ---
	name:	cache
	flags:	2
	size:	512 MiB 	0x20000000 B  
	end:	620 MiB 	0x26c00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
628 MiB --- offset: 0x27400000 B --- 658505728 B --- 1286144 s ---
	name:	env
	flags:	0
	size:	8 MiB 	0x800000 B  
	end:	636 MiB 	0x27c00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
644 MiB --- offset: 0x28400000 B --- 675282944 B --- 1318912 s ---
	name:	logo
	flags:	1
	size:	32 MiB 	0x2000000 B  
	end:	676 MiB 	0x2a400000 B  
  ---------------------
  ####  GAP 8 MiB  ####
684 MiB --- offset: 0x2ac00000 B --- 717225984 B --- 1400832 s ---
	name:	recovery
	flags:	1
	size:	32 MiB 	0x2000000 B  
	end:	716 MiB 	0x2cc00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
724 MiB --- offset: 0x2d400000 B --- 759169024 B --- 1482752 s ---
	name:	rsv
	flags:	1
	size:	8 MiB 	0x800000 B  
	end:	732 MiB 	0x2dc00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
740 MiB --- offset: 0x2e400000 B --- 775946240 B --- 1515520 s ---
	name:	tee
	flags:	1
	size:	8 MiB 	0x800000 B  
	end:	748 MiB 	0x2ec00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
756 MiB --- offset: 0x2f400000 B --- 792723456 B --- 1548288 s ---
	name:	crypt
	flags:	1
	size:	32 MiB 	0x2000000 B  
	end:	788 MiB 	0x31400000 B  
  ---------------------
  ####  GAP 8 MiB  ####
796 MiB --- offset: 0x31c00000 B --- 834666496 B --- 1630208 s ---
	name:	misc
	flags:	1
	size:	32 MiB 	0x2000000 B  
	end:	828 MiB 	0x33c00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
836 MiB --- offset: 0x34400000 B --- 876609536 B --- 1712128 s ---
	name:	instaboot
	flags:	1
	size:	512 MiB 	0x20000000 B  
	end:	1348 MiB 	0x54400000 B  
  ---------------------
  ####  GAP 8 MiB  ####
1356 MiB --- offset: 0x54c00000 B --- 1421869056 B --- 2777088 s ---
	name:	boot
	flags:	1
	size:	32 MiB 	0x2000000 B  
	end:	1388 MiB 	0x56c00000 B  
  ---------------------
  ####  GAP 8 MiB  ####
1396 MiB --- offset: 0x57400000 B --- 1463812096 B --- 2859008 s ---
	name:	system
	flags:	1
	size:	1536 MiB 	0x60000000 B  
	end:	2932 MiB 	0xb7400000 B  
  ---------------------
  ####  GAP 8 MiB  ####
2940 MiB --- offset: 0xb7c00000 B --- 3082813440 B --- 6021120 s ---
	name:	data
	flags:	4
	size:	11972 MiB 	0x2ec400000 B  
	end:	14912 MiB 	0x3a4000000 B  
====================================================================

**Comment by devmfc** [comment]
I think you have a X96 version with an S905W instead of an S905X. Can you post the output of `cat /sys/bus/soc/devices/soc0/* 2>/dev/null`?

**Reply by ratsalad** [reply]
Amlogic Meson
Amlogic Meson GXL (S905X) P212 Development Board
21:a - 82:2
GXL (S905X)

**Reply by devmfc** [reply]
No, it is an S905x.

Did you use your boot media (usb drive or sd card) on another box before?

**Reply by devmfc** [reply]
Run 'source /etc/boot-run-once/03-set_fw_env_config'
and try again.

**Reply by ratsalad** [reply]
I was using a microsd card, but switched to a flash drive and then I ran source /etc/boot-run-once/03-set_fw_env_config and then rest of the process worked just fine.  I appreciate your work and the time that you take to help.
